drivers/adc/adc_stm32.c

```diff
@@ -1005,4 +1114,8 @@ static void adc_stm32_cfg_func_##index(void)				\
 	irq_enable(DT_INST_IRQN(index));				\
 }
 
+#ifdef CONFIG_ADC_STM32_SHARED_IRQS
+DT_INST_FOREACH_STATUS_OKAY(STM32_ADC_INIT_SHARED)
+#else
 DT_INST_FOREACH_STATUS_OKAY(STM32_ADC_INIT)
+#endif
```

Comment:
Alternative otpion is to put the `#ifdef CONFIG_ADC_STM32_SHARED_IRQS` inside `STM32_ADC_INIT` definition, to minimize code changes